systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaUtils.java

```diff
@@ -645,4 +648,67 @@ public static void verifyKafkaKraftMetadataLog(final TestStorage testStorage, fi
     private static String buildDirectoryPath(int volumeId, int kafkaIndex) {
         return String.format("/var/lib/kafka/data-%d/kafka-log%d/__cluster_metadata-0", volumeId, kafkaIndex);
     }
+
+    /**
+     * Waits until the Kafka cluster in the given namespace the specified AutoRebalance state is set.
+     *
+     * @param namespaceName         The name of the Kubernetes namespace where the Kafka cluster resides.
+     * @param clusterName           The name of the Kafka cluster.
+     * @param expectedState         The expected {@link KafkaAutoRebalanceState} that is set.
+     */
+    public static void waitUntilKafkaHasAutoRebalanceState(final String namespaceName, final String clusterName,
+                                                           final KafkaAutoRebalanceState expectedState) {
+        TestUtils.waitFor(
+            String.format("waiting for Kafka cluster '%s' in namespace '%s' and KafkaRebalance state: '%s'", clusterName, namespaceName, expectedState),
+            TestConstants.GLOBAL_POLL_INTERVAL,
+            TestConstants.GLOBAL_TIMEOUT,
+            () -> {
+                Kafka kafka = KafkaResource.kafkaClient().inNamespace(namespaceName).withName(clusterName).get();
+                if (kafka == null || kafka.getStatus() == null) {
+                    return false; // Kafka cluster or its status is not available
+                }
+
+                final KafkaAutoRebalanceState currentState = kafka.getStatus().getAutoRebalance() != null
+                    ? kafka.getStatus().getAutoRebalance().getState()
+                    : null;
+
+                return currentState != null && currentState.equals(expectedState);
+            }
+        );
+    }
+
+    /**
+     * Waits until the Kafka cluster in the given namespace reaches the specified AutoRebalance mode and brokers.
+     *
+     * @param namespaceName                     The name of the Kubernetes namespace where the Kafka cluster resides.
+     * @param clusterName                       The name of the Kafka cluster.
+     * @param expectedKafkaAutoRebalanceMode    The expected {@link KafkaRebalanceMode} that the Kafka cluster should reach.
+     * @param expectedBrokers                   A list of expected broker IDs that the Kafka cluster should have in AutoRebalance state.
+     */
+    public static void waitUntilKafkaHasExpectedAutoRebalanceModeAndBrokers(final String namespaceName, final String clusterName,
```

Comment:
If you want to break it to multiple lines, I think that this:
```
public static void waitUntilKafkaHasExpectedAutoRebalanceModeAndBrokers(
    final String namespaceName, 
    final String clusterName,
    final KafkaRebalanceMode expectedKafkaAutoRebalanceMode,
    final List<Integer> expectedBrokers
) {
```
is clearer (and we use it throughout the code) - this applies for all other methods as well.

Reply:
Yeah make sense thanks.